pkg/manager/impl/validators/partition_validator.go

```diff
@@ -0,0 +1,83 @@
+package validators
+
+import (
+	"fmt"
+
+	"github.com/lyft/datacatalog/pkg/errors"
+	datacatalog "github.com/lyft/datacatalog/protos/gen"
+	"google.golang.org/grpc/codes"
+)
+
+const (
+	partitionKeyName   = "partitionKey"
+	partitionValueName = "partitionValue"
+)
+
+func ValidatePartitions(datasetPartitionKeys []string, artifactPartitions []*datacatalog.Partition) error {
+	if len(datasetPartitionKeys) != len(artifactPartitions) {
+		return errors.NewDataCatalogErrorf(codes.InvalidArgument, "Partition key mismatch, dataset keys: %+v, artifact Partitions: %+v", datasetPartitionKeys, artifactPartitions)
+	}
+
+	// Not all datasets need to be partitioned
+	if len(datasetPartitionKeys) == 0 && len(artifactPartitions) == 0 {
+		return nil
+	}
+
+	// compare the contents of the datasetkeys and artifact keys
+	partitionErrors := make([]error, 0)
+	keyMismatch := false
+
+	partitionKeyMatches := make(map[string]bool, len(artifactPartitions))
+	for _, datasetPartitionKey := range datasetPartitionKeys {
+		partitionKeyMatches[datasetPartitionKey] = false
+	}
+
+	for idx, artifactPartition := range artifactPartitions {
+		if artifactPartition == nil {
+			partitionErrors = append(partitionErrors, NewMissingArgumentError(fmt.Sprintf("%v[%v]", partitionKeyName, idx)))
+			continue
+		}
+
+		if err := ValidateEmptyStringField(partitionKeyName, artifactPartition.Key); err != nil {
+			partitionErrors = append(partitionErrors, NewMissingArgumentError(fmt.Sprintf("%v[%v]", partitionKeyName, idx)))
+		} else if err := ValidateEmptyStringField(partitionValueName, artifactPartition.Value); err != nil {
+			partitionErrors = append(partitionErrors, NewMissingArgumentError(fmt.Sprintf("%v[%v]", partitionValueName, idx)))
+		} else {
+			_, ok := partitionKeyMatches[artifactPartition.Key]
+
+			if ok {
+				partitionKeyMatches[artifactPartition.Key] = true
+			} else {
+				keyMismatch = true
```

Comment:
would it be helpful to record the specific keys that are mismatched in the collected err response?

Reply:
Yea we output the map of `partitionKeyMatches` in line 57 below